src/metabase/pulse.clj

```diff
@@ -1,50 +1,105 @@
 (ns metabase.pulse
-  (:require [hiccup.core :refer [html h]]
-            [clj-time.core :as t]
-            [clj-time.coerce :as c]
-            [clj-time.format :as f]
-            [clojure.java.io :as io]
+  (:require [clojure.java.io :as io]
+            (clojure [pprint :refer [cl-format]]
+                     [string :refer [upper-case]])
             [clojure.tools.logging :as log]
-            [clojure.pprint :refer [cl-format]]
-            [clojure.string :refer [upper-case]]
-            [metabase.models.setting :as setting]))
+            (clj-time [coerce :as c]
+                      [core :as t]
+                      [format :as f])
+            [hiccup.core :refer [html h]]
+            [metabase.models.setting :as setting]
+            [metabase.util :as u])
+  (:import (java.awt BasicStroke
+                     Color
+                     RenderingHints)
+           java.awt.image.BufferedImage
+           java.io.ByteArrayOutputStream
+           java.util.Date
+           javax.imageio.ImageIO
+           org.apache.commons.io.IOUtils
+           (org.fit.cssbox.css CSSNorm
+                               DOMAnalyzer
+                               DOMAnalyzer$Origin)
+           (org.fit.cssbox.io DefaultDOMSource
+                              StreamDocumentSource)
+           org.fit.cssbox.layout.BrowserCanvas
+           org.fit.cssbox.misc.Base64Coder))
 
 ;; NOTE: hiccup does not escape content by default so be sure to use "h" to escape any user-controlled content :-/
 
 ;;; ## CONFIG
 
-(def ^:const card-width 400)
-(def ^:const rows-limit 10)
-(def ^:const cols-limit 3)
-(def ^:const sparkline-dot-radius 6)
-(def ^:const sparkline-thickness 3)
-(def ^:const sparkline-pad 8)
+(def ^:private ^:const card-width 400)
```

Comment:
things that are only used in the current namespace should be `^:private` especially if they are undocumented